### ISSUE TITLE
file cache is not getting saved

### ISSUE THREAD

**Issue by mohsininayatkhan** [ankitpokhrel/tus-php#73]
Using tus client for uploading a file and getting following error in http error log

_PHP Fatal error:  Uncaught TypeError: Argument 2 passed to TusPhp\\File::seek() must be of the type integer, null given, called in /var/www/html-tus/vendor/ankitpokhrel/tus-php/src/Tus/Client.php on line 527 and defined in /var/www/html-tus/vendor/ankitpokhrel/tus-php/src/File.php:387_

While debugging, found that issue can be in the following lines of code of 'getData' function

```php
$file   = new File;
$handle = $file->open($this->getFilePath(), $file::READ_BINARY);
$offset = $this->partialOffset;

if ($offset < 0) {
    $fileMeta = $this->getCache()->get($key);
    $offset   = $fileMeta['offset'];
}

$file->seek($handle, $offset);
```

As $offset value is -1 and when it tries to get information from cache by key, it returns null.

Using file as cache and below are my configurations for cache from  /vendor/ankitphokhrel/tus-php/src/Config/default.php

```php
/**
 * File cache configs.
*/
'file' => [
     'dir' => '/var/www/html-tus/cache/',
     'name' => 'tus_php.cache',
],
```

**Please also note that cache directory have write privileges and I have created 'tus_php.cache'  by myself. However, no content get written in this during file upload**

**Comment by ankitpokhrel** [comment]
@mohsininayatkhan can you share details how you setup the server? It looks like you are using file cache in client but redis in server.

**Comment by mohsininayatkhan** [comment]
The server setup is on different machine and is written in C#. It will not work with different server?

**Comment by ankitpokhrel** [comment]
The client gets [upload offset](https://github.com/ankitpokhrel/tus-php/blob/master/src/Tus/Client.php#L478) from cache while sending upload request to the server and only sends the chunk of data that hasn't been uploaded yet instead of sending whole file again. So, the client should be able to access cache set by the server as per current implementation. 

But it seems that client doesn't really need to be dependent on the cache. Will check more on how we can remove this dependency.

**Comment by ankitpokhrel** [comment]
Fixed in #76
